File: AtCoder/ACP/BootCamp4b/Medium100_2/python/A_Prefix_and_Suffix.py
N = int(input())
s = input()
t = input()

res = N
for i in range(N):
    f = True
    for j in range(i, N):
        if s[j] != t[j - i]:
            f = False
            break
    if f:
        break
    res += 1
print(res)

File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/A_Prefix_and_Suffix.cpp
#include <bits/stdc++.h>
using namespace std;

int N;
string s, t;

void solve() {
  int res = N;
  for (int i = 0; i < N; ++i) {
    bool f = true;
    for (int j = i; j < N; ++j) {
      if (s[j] != t[j - i]) {
        f = false;
        break;
      }
    }
    if (f) {
      break;
    }
    ++res;
  }
  printf("%d\n", res);
}

int main() {
  cin >> N >> s >> t;
  solve();
}